File: calculator.pl
use feature qw(switch); # needed for given($option) althought deprecated

sub calculator{
  print "***************************\n";
  print "A SIMPLE CALCULATOR IN PERL\n";
  print "ENTER INPUTS WITHOUT SPACES\n";
  print "***************************\n\n\n";
  
  while(1){
    print "Enter an operation (+, -, *, /) | 'q' to quit: ";
    my $option = <STDIN>;
    chomp($option); # remove new line
    if ($option eq 'q' || $option eq 'Q'){
      last;
    }
    print "Enter number 1 (double): ";
    my $num_1 = <STDIN>;
    print "Enter number 2 (double): ";
    my $num_2 = <STDIN>;
    my $total;
    
    given($option){
      when('+'){
        $total = $num_1 + $num_2;
      }
      when('-'){
        $total = $num_1 - $num_2;
      }
      when('*'){
        $total = $num_1 * $num_2;
      }
      when('/'){
        $total = $num_1 / $num_2;
      }
    }
    print "The result is: " . $total . "\n";
  }
}

calculator()

File: calculator.rb
class Calculator # doesn't need a class
    def main # doesn't need a function
        puts "***************************"
        puts "A SIMPLE CALCULATOR IN RUBY"
        puts "ENTER INPUTS WITHOUT SPACES"
        puts "***************************\n\n"

    while true
        print "Enter an operation (+, -, *, /) | 'q' to quit: "
        option = gets[0]
        break if option == 'q' || option == 'Q'
        print "Enter number 1 (double): "
        num_1 = gets.chomp.to_f # chomp to remove new line, to_f to parse to a float (8-byte)
        print "Enter number 2 (double): "
        num_2 = gets.chomp.to_f
        @total  # Class attribute is unecessary, didn't need a class either I just felt like showing some syntax
        case option
        when '+'
            total = num_1  + num_2 
        when '-'
            total = num_1  - num_2 
        when '*'
            total = num_1  * num_2
        when '/'
            total = num_1  / num_2
        end
        puts "The result is: #{total}"
    end
    end
end

Calculator.new.main # create a new instance of Calculator and call main
